File: 0447-number-of-boomerangs/0447-number-of-boomerangs.py
class Solution:
    def numberOfBoomerangs(self, points):
        result = 0

        for i in range(len(points)):
            count = {}

            for j in range(len(points)):
                if i == j:
                    continue

                dx = points[i][0] - points[j][0]
                dy = points[i][1] - points[j][1]

                distance = dx * dx + dy * dy

                count[distance] = count.get(distance, 0) + 1

            for freq in count.values():
                result += freq * (freq - 1)

        return result

File: 0447-number-of-boomerangs/0447-number-of-boomerangs.cpp
class Solution {
public:
    int numberOfBoomerangs(vector<vector<int>>& points) {
        int n = points.size();
        int ans = 0;

        for (int i = 0; i < n; i++) {

            unordered_map<long long, int> mp;

            for (int j = 0; j < n; j++) {
                if (i == j) continue;

                long long dx = points[i][0] - points[j][0];
                long long dy = points[i][1] - points[j][1];

                long long dist = dx * dx + dy * dy;

                mp[dist]++;
            }

            for (auto& [dist, cnt] : mp) {
                ans += cnt * (cnt - 1);
            }
        }

        return ans;
    }
};